dlt/destinations/snowflake/snowflake.py

```diff
@@ -48,44 +51,76 @@
     "VARIANT": "complex"
 }
 
+
 class SnowflakeLoadJob(LoadJob, FollowupJob):
     def __init__(
             self, file_path: str, table_name: str, write_disposition: TWriteDisposition, load_id: str, client: SnowflakeSqlClient,
-            stage_name: Optional[str] = None, keep_staged_files: bool = True
+            stage_name: Optional[str] = None, keep_staged_files: bool = True, forward_staging_credentials: bool = True, staging_config: Optional[DestinationClientStagingConfiguration] = None
     ) -> None:
         file_name = FileStorage.get_file_name_from_file_path(file_path)
         super().__init__(file_name)
 
         with client.with_staging_dataset(write_disposition == "merge"):
             qualified_table_name = client.make_qualified_table_name(table_name)
 
-            if stage_name:
+            # extract and prepare some vars
+            bucket_path = CopyFileLoadJob.get_bucket_path(file_path) if NewReferenceJob.is_reference_job(file_path) else ""
+            file_name = FileStorage.get_file_name_from_file_path(bucket_path) if bucket_path else file_name
+            from_clause = ""
+            credentials_clause = ""
+            files_clause = ""
+            stage_file_path = ""
+
+            # create storage integration stage for reference jobs if defined, for now gcs only
+            if bucket_path and stage_name:
+                from_clause = f"FROM @{stage_name}/"
+                # select all files that match the file from our job
+                bucket = CopyFileLoadJob.get_bucket(file_path)
+                file_path = bucket_path.replace(bucket + "/", "")
+                files_clause = f"FILES = ('{file_path}')"
+            # s3 credentials case
+            elif bucket_path:
+                if bucket_path.startswith("s3://") and forward_staging_credentials:
+                    aws_access_key = staging_config.credentials.aws_access_key_id # type: ignore
+                    aws_secret_key = staging_config.credentials.aws_secret_access_key # type: ignore
+                    credentials_clause = f"""CREDENTIALS=(AWS_KEY_ID='{aws_access_key}' AWS_SECRET_KEY='{aws_secret_key}')"""
+                from_clause = f"FROM '{bucket_path}'"
+            # create a stage if so defined
+            elif stage_name:
                 # Concat "SCHEMA_NAME".stage_name
                 stage_name = client.make_qualified_table_name(stage_name)
                 # Create the stage if it doesn't exist
                 client.execute_sql(f"CREATE STAGE IF NOT EXISTS {stage_name}")
+            # or use implicit stage
             else:
                 # Use implicit table stage by default: "SCHEMA_NAME"."%TABLE_NAME"
                 stage_name = client.make_qualified_table_name('%'+table_name)
 
+            if not bucket_path: # this means we have a local file
```

Comment:
theoretically someone can use `file://` with staging. do you think it makes sense to handle it?